### PR TITLE
Autoconvert assets if browser supports it

### LINKED ISSUE #16068: tif image conversion
### Describe the Bug

TIF images are rendered as black in Directus App File Browser.

Screenshot 2022-10-20 at 07 24 37

Here are the request made by Directus:
https://db.dl.tlu.ee/assets/a0928c43-315e-4631-be4c-f61ffa0f0c1d?key=system-medium-cover&modified=2022-10-20T04:21:12.000Z&download=true
https://db.dl.tlu.ee/assets/c0d2ac87-e167-45e0-a682-e69cb318a9d0?key=system-medium-cover&modified=2022-10-20T04:21:14.000Z&download=true

Same image can be dispayed using the trasformation setup i have created for thumnails:
https://db.dl.tlu.ee/assets/a0928c43-315e-4631-be4c-f61ffa0f0c1d?key=thumbnail
https://db.dl.tlu.ee/assets/c0d2ac87-e167-45e0-a682-e69cb318a9d0?key=thumbnail

I have followed instructios (#16046) and set the ASSETS_TRANSFORM_IMAGE_MAX_DIMENSION to 11000. Also checked the mime types. 

### To Reproduce

Upload TIF image via Directus App File Browser.
You will see black thumbnail instead of the image.
Test images i used:
Medium size - https://www.icloud.com/iclouddrive/0de-79pzgV2rPBQm3rk1BcRAQ#medium_10084x8146
Small size - https://www.icloud.com/iclouddrive/061osRRDOYaI4vyfDrMKB11Tw#small_4000x3231

### Errors Shown

Tested with the lates Brave browser (Version 1.44.112 Chromium: 106.0.5249.119 (x86_64)), and Safari (Version 16.0 (16614.1.25.9.10, 16614)). No other errors except in Safari  "Refused to execute a script because its hash, its nonce, or 'unsafe-inline' does not appear in the script-src directive of the Content Security Policy." 

### What version of Directus are you using?

9.18.1

### What version of Node.js are you using?

14.18.2

### What database are you using?

Maria DB

### What browser are you using?

Brave browser (Version 1.44.112 Chromium: 106.0.5249.119 (x86_64)), and Safari (Version 16.0 (16614.1.25.9.10, 16614))

### How are you deploying Directus?

Directus runs at Uni server, set up manually

### PR DESCRIPTION
**Pull request by Nitwel**
Fixes #16068
Fixes ENG-157
Ref: #15918

### CODE REVIEW

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
Lets update this to be a `?format=auto` query param instead. I don’t think we need it to be an environment variable

**Comment by Nitwel** [comment]
Updated it to a query option.

**Comment by paescuj** [comment]
Added a quick & dirty (but so far successful) approach in [7922e1c](https://github.com/directus/directus/pull/18012/commits/7922e1cd6f61924d5ad775e37d4fafab1c5f75a2) to auto transform images within the app as per https://github.com/directus/directus/issues/16068#issuecomment-1285849927.

@Nitwel Could you verify if the idea of allowing the `format=auto` option along with the `key` option makes sense and if so clean this pull request up so we can get it over the line?
Thanks! 😃

**Comment by Nitwel** [comment]
It doesn't make sense to allow for `format=auto` when having a key set, as a key references a preset that contains the formatting. Meaning, we should change the presets instead of hacking our way around it. 😄 

But you brought up a good point which is that I did forgot a presets implementation for assets. Added that now.

**Comment by rijkvanzanten** [comment]
@Nitwel @paescuj This is looking very promising so far! What's left TBD to get this review-ready / merged? 🙂

**Comment by Nitwel** [comment]
Only a quick lint run 😅 and some testing but I'm happy where it's at right now.

File: api/src/types/assets.ts [hunk 1]
```diff
@@ -65,21 +65,11 @@ export type TransformationMap = {
...
-};
-
-// Transformation preset is defined in the admin UI.
-export type TransformationPreset = TransformationPresetFormat &
-	TransformationPresetResize &
-	TransformationParams & { key: string };
-
-export type TransformationPresetFormat = {
-	format?: 'jpg' | 'jpeg' | 'png' | 'webp' | 'tiff' | 'avif';
+	format?: 'auto' | 'jpg' | 'jpeg' | 'png' | 'webp' | 'tiff' | 'avif';
```
Comment (paescuj) [inline api/src/types/assets.ts:73]: Small note on the side: While `auto` is now a valid input for presets on the user side, it must not be passed to `resolvePreset()`

**Review by rijkvanzanten — APPROVED** [review]

**Review by paescuj — APPROVED** [review]
Thanks @Nitwel  for giving it the finishing touch 😄
This is a great addition, especially from the app side 👍 
Nice clean-up, all my tests succeeded!

**Pull request merged**